### ISSUE TITLE
Create orchestration layer for async projections operations in .NET client

### ISSUE THREAD

**Issue by jen20** [EventStore/EventStore#672]
When issuing queries, some plumbing is required to know when it is complete - the `Async` in the method names refers to the HTTP calls rather than the operations themselves. As such it might be better to provide a higher level API that presents queries as a `Task<QueryResult>` or whatever, and polls the status/gets the state internally.

This set of posts on the list describes this in more detail: https://groups.google.com/forum/#!topic/event-store/LAfnAYibtIc

**Comment by riccardone** [comment]
I understand this requirement more or less .
I wonder if you would like to attach stories to any new requirement before starting to code it. 
As an example...

As a developer 
I want an API that exposes asynchronous Tasks around HTTP calls for projection operations 
So that I can wait and get a QueryResult

some acceptance criteria..

Given that I need to query a projection
When I create an instance of a new Projection Api
Then I expect to see Task<QueryResult>

Given that I create an instance of (to be defined) Api
When I call (tbd) async method
Then I expect to receive (tbd) as a result

...

**Comment by pgermishuys** [comment]
If the requirements aren't clear, whoever wants to pick it up can always start a conversation in the issue or the PR even to discuss the development requirements and if stories helps both parties understand what needs to be done, all the better.

**Comment by jasonmitchell** [comment]
I would be interested in picking this one up - I think I understand the issue.  I'm going to create a test to recreate the issue and then I may be back with some questions.

**Comment by jasonmitchell** [comment]
I've got a basic implementation of this which introduces a `QueryManager` class with a method `GetResultAsync` which will internally poll for a status of Completed on the query with an increasing delay to avoid polling too regularly on long running query.  I've also added a timeout for getting query results which will cancel the polling if it is taking too long.  

Should I open an in-progress PR for discussion?
